jdaviz/components/toolbar_nested.py

```diff
@@ -53,13 +65,74 @@ def __init__(self, viewer, tools_nested, default_tool_priority=[]):
         self.default_tool_priority = default_tool_priority
         self._handle_default_tool()
 
+        for msg in (AddDataMessage, RemoveDataMessage, ViewerAddedMessage, ViewerRemovedMessage,
+                    SpectralMarksChangedMessage):
+            self.viewer.hub.subscribe(self, msg,
+                                      handler=lambda _: self._update_tool_visibilities())
+
+    def _is_visible(self, tool_id):
+        # tools can optionally implement self.is_visible(). If not NotImplementedError
+        # the tool will always be visible
+        return getattr(self.tools[tool_id], 'is_visible', lambda: True)()
+
+    def _update_tool_visibilities(self):
+        needs_deactivate_active = False
+        for menu_ind in range(self._max_menu_ind):
+            has_primary = False
+            nvisible = 0
```

Comment:
"nvisible" almost reads like "invisible", which is a little confusing. I propose you add underscore.
```suggestion
            n_visible = 0
```